app/src/main/java/com/example/instagram/fragments/ProfileFragment.java

```diff
@@ -0,0 +1,170 @@
+            user = Parcels.unwrap(getArguments().getParcelable(MainActivity.KEY_USER));
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        fragmentProfileBinding = FragmentProfileBinding.inflate(inflater, container, false);
+        return fragmentProfileBinding.getRoot();
+    }
+
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        ivProfileImg = fragmentProfileBinding.ivProfileImg;
+        tvUsername = fragmentProfileBinding.tvUsername;
+        gvPosts = fragmentProfileBinding.gvPosts;
+        fabLogOut = fragmentProfileBinding.fabLogout;
+
+        tvUsername.setText(user.getUsername());
+
+        ParseFile profileImg = user.getParseFile("profileImg");
+        if (profileImg != null) {
+            Glide.with(getActivity()).load(profileImg.getUrl().replaceAll("http", "https")).circleCrop().into(ivProfileImg);
+        } else {
+            Log.e(TAG, "CANNOT FIND IMAGE");
+        }
+
+        // Allow user to update avatar by clicking on current profile picture
+        ivProfileImg.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                goToCompose();
+            }
+        });
+
+        fabLogOut.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ParseUser.logOutInBackground(new LogOutCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e != null) {
+                            Log.i(TAG, String.valueOf(e));
+                            return;
+                        }
+                        Log.i(TAG, "Logged out!");
+                        Toast.makeText(getContext(), "Logged out successfully", Toast.LENGTH_SHORT).show();
+                        goLogIn();
+                    }
+                });
+            }
+        });
+
+        posts = new ArrayList<>();
+        adapter = new PostGridAdapter(getActivity(), posts);
+        gvPosts.setAdapter(adapter);
+        queryPosts();
+    }
+
+    // Go to compose & send information of current user to update avatar (instead of composing post)
+    private void goToCompose() {
+        ComposeFragment composeFragment = ComposeFragment.newInstance(Parcels.wrap(user));
+        MainActivity.fragmentManager.beginTransaction().replace(R.id.flContainer, composeFragment).commit();
+    }
+
+    private void queryPosts() {
+        MainActivity.showProgressBar();
+        Log.i(TAG, "Start querying for new post in profile");
+
+        // Set up query
+        ParseQuery<Post> query = ParseQuery.getQuery(Post.class);
+        query.include(Post.KEY_USER);
+        query.addDescendingOrder("createdAt");
+        query.whereContains("user", user.getObjectId());
```

Comment:
Same thing as in previous files, you can use keys instead of hardcoding strings (in case you end up changing the names of columns in Parse)